### PR TITLE
BAH-1685 | Report upload against multiple tests

### PR DESCRIPTION
**Pull request by Arjun-Go**
## Requirements

- [x] This PR has a proper title that briefly describes the work done
- [x] I have squashed / amended the comments to make it more redable
- [x] I have included link to all the JIRA ticket('s)
- [x] I wrote the code as a pair or atleast performed a self-review of my own code
- [x] I have updated the documentation on [Bahmni Wiki](https://bahmni.atlassian.net/wiki/spaces/BAH/overview) or [README](https://github.com/Bahmni/bahmni-lab-frontend/blob/main/README.md) (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary
Allows user to add multiple tests for report upload. Multiple tests will be shown in the reports table as comma separated values.

## Screenshots
image

## JIRA tickets
https://bahmni.atlassian.net/jira/software/c/projects/BAH/boards/35?modal=detail&selectedIssue=BAH-1685

### CODE REVIEW

File: src/report-table/report-table.component.tsx [hunk 1]
```diff
@@ -172,4 +175,27 @@ const ReportTable = props => {
...
+      if (existingReports.length === 1) {
+        existingReports[0].resource.code.coding[0].display = `${existingReports[0].resource.code.coding[0].display}, ${diagnosticReport.resource.code.coding[0].display}`
+      }
+    } else {
+      documentUrls.push(documentUrl)
+      uniqueDiagnosticReports.push(diagnosticReport)
+    }
+  }
+  return uniqueDiagnosticReports
+}
```
Comment (N0-man) [inline src/report-table/report-table.component.tsx:202]: While the code is imperative, it still needed eyeballing up and down to get the gist of what we are trying to do here... you can refactor it to be more functional with map-reduce

1) Find unique report references - dont merge this with the rest of deduplication code
2) Your unique report references (url) would be the guide towards deduplication (map). For every unique report reference (url) accumulate the code (reduce). 
3) lastly you are deduplicating - this name sounds more like what function is litrally doing  `mergeMultipleTestsBasedOnUploadedFile`

```
function url(resource: Resource) {
  return resource.presentedForm[0].url
}
function code(resource: Resource) {
  return resource.code.coding[0].display
}

function getUniqueReportUrls(diagnosticReport: ReportEntry[]) {
  return new Set<string>(diagnosticReport.map(report => url(report.resource)))
}

function dedupe(diagnosticReport: Array<ReportEntry>) {
  return Array.from(getUniqueReportUrls(diagnosticReport)).map(reportUrl =>
    diagnosticReport.reduce<ReportEntry | undefined>((acc, curr) => {
      if (url(curr.resource) === reportUrl) {
        if (acc)
          acc.resource.code.coding[0].display = `${code(acc.resource)}, ${code(
            curr.resource,
          )}`
        else return curr
      }
      return acc
    }, undefined),
  )
}
```

File: src/upload-report/upload-report.tsx [hunk 2]
```diff
@@ -146,3 +146,35 @@ const UploadReport: React.FC<UploadReportProps> = ({
...
+      reportConclusion,
+      ac,
+      selectedPendingOrder,
+    )
+    if (allSuccess && !diagnosticReportResponse.ok) {
+      allSuccess = false
+    }
+  }
+  if (allSuccess) {
+    close(true)
```
Comment (N0-man) [inline src/upload-report/upload-report.tsx:178]: what if some test upload fails?

File: src/report-table/report-table.component.tsx (hunk 1 above)
Comment (Arjun-Go) [reply src/report-table/report-table.component.tsx:202]: changes have been made

File: src/upload-report/upload-report.tsx (hunk 2 above)
Comment (Arjun-Go) [reply src/upload-report/upload-report.tsx:178]: Currently we are not handling the failure scenario. A new card will be added in the backlog by Abhinab.

**Review by N0-man — APPROVED** [review]

**Pull request merged**